app/services/hyrax/listeners/metadata_index_listener.rb

```diff
@@ -35,9 +40,16 @@ def on_object_deleted(event)
 
       private
 
-      def log_non_resource(event)
+      def log_non_resource(event, idx = :object)
         Hyrax.logger.info('Skipping object reindex because the object ' \
-                          "#{event[:object]} was not a Valkyrie::Resource.")
+                          "#{event[idx]} was not a Valkyrie::Resource.")
+      end
+
+      def metadata_updated(event, idx)
+        log_non_resource(event, idx) && return unless
```

Comment:
does this evaluate right? it looks like it will `log_non_resource` in all cases, and then `return unless...`?

Reply:
I just copied this down, so I assumed it is correct.  It may get re-evaluated as part of the requested method signature changes.

Reply:
This is a normal unless guard.  The log and return will not happen `unless` the resource isn't a Valkyrie::Resource.  It doesn't scan well.  I could put parentheses around the log && return part for readability.

Reply:
I did a refactor to make this more readable.  The check is simplified down to...
```
return unless resource? {resource}
```

The logging now occurs in the method checking if the payload has a Valkyrie::Resource.